File: src/main/java/sage/web/context/JsTemplate.java
package sage.web.context;

import java.io.File;
import java.io.IOException;
import java.io.UncheckedIOException;
import java.nio.file.Files;

public class JsTemplate {
  private File file;
  private String source;
  private long loadedModifiedTime;

  public JsTemplate(File file) {
    this.file = file;
    load();
  }

  public String getSource() {
    if (ComponentResourceManager.instance().isDevelopmentMode()) {
      if (file.lastModified() != loadedModifiedTime) {
        load();
      }
    }
    return source;
  }

  private void load() {
    try {
      source = new String(Files.readAllBytes(file.toPath()), "UTF-8");
    } catch (IOException e) {
      throw new UncheckedIOException(e);
    }
    loadedModifiedTime = file.lastModified();
  }
}

File: src/main/kotlin/sage/web/context/JsTemplate.kt
package sage.web.context

import java.io.File
import java.nio.charset.StandardCharsets
import java.nio.file.Files

open class JsTemplate(private val file: File) {

  var source: String = ""
    get() {
      if (file.lastModified() != loadedModifiedTime) {
        load()
      }
      return field
    }
    private set

  private var loadedModifiedTime: Long = 0

  init {
    load()
  }

  private fun load() {
    source = Files.readAllBytes(file.toPath()).toString(StandardCharsets.UTF_8)
    loadedModifiedTime = file.lastModified()
  }
}
